File: Queue and Stack/Palindrome.py
class Palindrome:
	def __init__(self):
		self.stack = []
		self.queue = []

	def enqueue(self, s):
		for char in s:
			self.queue.append(char)

	def pushStack(self, s):
		for char in s:
			self.stack.append(char)

	def dequeue(self):
		return self.queue.pop(0)

	def popStack(self):
		return self.stack.pop()

	def check(self, s):
		s = s.lower()
		self.enqueue(s)
		self.pushStack(s)
		for i in range(len(s)):
			if self.dequeue() != self.popStack():
				self.stack = []
				self.queue = []
				return False
		return True

pal = Palindrome()
print(pal.check("Palindrome"))
print(pal.check("aBccba"))
print(pal.check("abcba"))

File: Queue and Stack/Palindrome.js
function Palindrome(){
	this.stack = [];
	this.queue = [];
}

Palindrome.prototype.fillStackQueue = function(s){
	for(var i=0; i<s.length; i++){
		this.queue.push(s[i]);
		this.stack.push(s[i]);
	}
}

Palindrome.prototype.dequeue = function(){
	return this.queue.shift();
}

Palindrome.prototype.stackPop = function(){
	return this.stack.pop();
}

Palindrome.prototype.check = function(s){
	s = s.toLowerCase();
	this.fillStackQueue(s);
	for(var i=0; i<s.length; i++){
		if(this.dequeue() !== this.stackPop()){
			this.stack = [];
			this.queue = [];
			return false;
		}
	}
	return true;
}

var pal = new Palindrome();
console.log(pal.check("Palindrome"));
console.log(pal.check("Madam"));
console.log(pal.check("rotator"));